shaders/src/main.fs

```diff
@@ -66,6 +66,26 @@ void main() {
     fragColor = fog(fragColor, view);
 #endif
 
+#if defined(VARIANT_HAS_SHADOWING) && defined(VARIANT_HAS_DIRECTIONAL_LIGHTING)
+    if (CONFIG_DEBUG_DIRECTIONAL_SHADOWMAP) {
+        float a = fragColor.a;
+        highp vec4 p = getShadowPosition(getShadowCascade());
+        p.xy = p.xy * (1.0 / p.w);
+        if (p.xy != saturate(p.xy)) {
+            vec4 c = vec4(1.0, 0, 1.0, 1.0) * a;
+            fragColor = mix(fragColor, c, 0.2);
+        } else {
+            p.xy = saturate(p.xy);
```

Comment:
isn't this already true if we're in this conditional block?

Reply:
yes! good catch.